ethereum/contracts/Bridge.sol

```diff
@@ -61,6 +57,8 @@ contract Bridge {
         require(codehash != 0x0, "There's no account for this address on the network");
         require(codehash != NON_CONTRACT_ACCOUNT_HASH, "Only contract accounts can be registered as applications");
 
-        return true;
+        Application app = Application(_appID);
+        require(app.register(address(this)), "Application failed registration");
```

Comment:
`app.register` always returns true, so the `require` seems unnecessary.

Reply:
in our current apps yes, but may be useful for future apps to allow failing registration?

Reply:
- Vincent: Already registered applications will throw an error here on `app.register(address(this))` due to the check `require(bridge == address(0), "Bridge has already been registered");` so it's redundant.
- Aidan: Future apps should fail registration if they're already registered, which is handled by the current code.

I've updated the contracts in https://github.com/Snowfork/polkadot-ethereum/pull/115/commits/ce891d0610f5a38410cd1ff327087dda266cc932